packages/react/modules/components/configure-adapter/with-reconfiguration.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import type { Adapter } from '@flopflip/types';
+
+import React, { PureComponent, type ComponentType, type Node } from 'react';
+import { wrapDisplayName } from 'recompose';
+import { AdapterContext } from './configure-adapter';
+
+type RequiredProps = {};
+type ProvidedProps = {};
+
+const withReconfiguration = (propKey: string = 'reconfigure') => (
+  Component: ComponentType<RequiredProps>
+) => {
+  class EnhancedComponent extends PureComponent<
+    $Diff<RequiredProps, ProvidedProps>
+  > {
+    static displayName = wrapDisplayName(Component, 'withReconfiguration');
+    render(): Node {
+      return (
+        <AdapterContext.Consumer>
+          {reconfigure => (
+            <Component {...{ [propKey]: reconfigure }} {...this.props} />
```

Comment:
Shouldn't it be the other way around? Otherwise `...this.props` might accidentally override `[propKey]`

Reply:
Yeah I can change that true.